core/markets/network/impl/libp2p_market_network.cpp

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "markets/network/impl/libp2p_market_network.hpp"
+
+namespace fc::markets::network {
+  using libp2p::connection::Stream;
+  using libp2p::peer::PeerInfo;
+
+  Libp2pMarketNetwork::Libp2pMarketNetwork(
+      std::shared_ptr<Host> host)
+      : host_{std::move(host)} {}
+
+  void Libp2pMarketNetwork::newStream(
+      const PeerInfo &peer_info,
+      const Protocol &protocol,
+      const CborStreamResultHandler &handler) {
+    host_->newStream(peer_info, protocol, [handler](auto stream) {
+      if (stream.has_error()) {
+        handler(stream.error());
+        return;
+      }
+      handler(std::make_shared<CborStream>(stream.value()));
+    });
+  }
+
+  outcome::result<void> Libp2pMarketNetwork::setDelegate(
+      const Protocol &protocol,
+      const std::function<libp2p::connection::Stream::Handler> &handler) {
+    host_->setProtocolHandler(protocol, handler);
+    return outcome::success();
+  }
+
+  outcome::result<void> Libp2pMarketNetwork::stopHandlingRequests() {
+    // TODO (a.chernyshov) remove protocol handlers - not supported by
```

Comment:
This either lacks jira task id or should be removed. Otherwise, it is not clear what to do with this (unused?) method.

Reply:
Removed after consultation with cpp-libp2p developers.
No sence in sopping handling.